server.go

```diff
@@ -1369,6 +1369,13 @@ type Point struct {
 // WriteSeries writes series data to the database.
 // Returns the messaging index the data was written to.
 func (s *Server) WriteSeries(database, retentionPolicy string, points []Point) (uint64, error) {
+	// Make sure every point has at least one field.
+	for _, p := range points {
+		if len(p.Fields) == 0 {
+			return 0, fmt.Errorf("point %s has no fields", p.Name)
```

Comment:
Also, why not do this at line 1403? We are already walking the points there. Let's not walk at the top, then walk again.